setup_routing.go

```diff
@@ -90,12 +108,50 @@ func setupDelegatedRouting(cfg Config, dnsCache *cachedDNS) ([]routing.Routing,
 	return delegatedRouters, nil
 }
 
+// parseBootstrapPeers parses a list of bootstrap peer strings into AddrInfo structs
+// It skips empty strings and "auto" placeholders, and logs warnings for invalid peers
+func parseBootstrapPeers(peers []string, warnOnAuto bool) ([]peer.AddrInfo, error) {
+	bootstrapPeers := make([]peer.AddrInfo, 0, len(peers))
+	for _, peerStr := range peers {
+		if peerStr == "" {
+			continue
+		}
+		if peerStr == autoconf.AutoPlaceholder {
+			if warnOnAuto {
+				goLog.Warnf("Bootstrap peer 'auto' placeholder not expanded - is autoconf enabled?")
+			}
+			continue
+		}
+		ai, err := peer.AddrInfoFromString(peerStr)
+		if err != nil {
+			goLog.Warnf("Failed to parse bootstrap peer %q: %v", peerStr, err)
+			continue
+		}
+		bootstrapPeers = append(bootstrapPeers, *ai)
+	}
+	return bootstrapPeers, nil
+}
+
 func setupDHTRouting(ctx context.Context, cfg Config, h host.Host, ds datastore.Batching, dhtRcMgr network.ResourceManager, bwc metrics.Reporter) (routing.Routing, error) {
 	if cfg.DHTRouting == DHTOff {
 		return nil, nil
 	}
 
-	var err error
+	// Parse bootstrap peers
+	bootstrapPeers, err := parseBootstrapPeers(cfg.Bootstrap, true)
+	if err != nil {
+		return nil, err
+	}
+
+	// If no bootstrap peers provided, use defaults for seed peering or error otherwise
+	if len(bootstrapPeers) == 0 {
+		if cfg.SeedPeering {
+			// Use default bootstrap peers for seed peering
+			bootstrapPeers = dht.GetDefaultBootstrapPeerAddrInfos()
+		} else {
+			return nil, fmt.Errorf("no valid bootstrap peers configured - provide bootstrap peers or enable autoconf")
+		}
```

Comment:
[Indent error flow](https://go.dev/wiki/CodeReviewComments#indent-error-flow)
```suggestion
		if !cfg.SeedPeering {
			return nil, fmt.Errorf("no valid bootstrap peers configured - provide bootstrap peers or enable autoconf")
		}
		// Use default bootstrap peers for seed peering
		bootstrapPeers = dht.GetDefaultBootstrapPeerAddrInfos()
```